app/core/steps/EligibilityValidationStep.js

```diff
@@ -44,33 +45,40 @@ class EligibilityValidationStep extends ValidationStep {
         eligibilityCookie.setCookie(req, res, nextStepUrl, fieldKey, fieldValue);
     }
 
-    previousQuestionsAnswered(req, ctx, fieldKey) {
-        let eligibilityQuestionsList;
+    previousQuestionsAnswered(req, ctx, currentScreener) {
 
-        if (Object.keys(config.eligibilityQuestionsProbate).includes(fieldKey)) {
-            eligibilityQuestionsList = config.eligibilityQuestionsProbate;
-        } else if (Object.keys(config.eligibilityQuestionsIntestacy).includes(fieldKey)) {
-            eligibilityQuestionsList = config.eligibilityQuestionsIntestacy;
-        }
+        const isIntestacyScreener = Object.keys(config.intestacyScreeners).includes(currentScreener);
+        let screenersList = isIntestacyScreener ? config.intestacyScreeners : config.probateScreeners;
```

Comment:
Could potentially reduce conditional logic by moving out getScreeners to another method

Reply:
Done with my last commit :)